aodndata/gsla/handler.py

```diff
@@ -0,0 +1,159 @@
+                )
+
+            netcdf_file = netcdf_collection[0]
+            result_previous_version_creation_date = self.get_previous_version_creation_date(netcdf_file.name)
+
+            push_new_file = True  # by default we push to the storage the file landed in the pipeline
+            remove_previous_version = False
+
+            # check creation date of file already on storage
+            if result_previous_version_creation_date:
+                new_file_creation_date = get_creation_date(netcdf_file.name)
+                if result_previous_version_creation_date > new_file_creation_date:
+                    push_new_file = False
+                else:
+                    remove_previous_version = True
+                    previous_file_object = self.get_previous_version_object(netcdf_file.name)
+
+            if push_new_file:
+                # index un-gzipped file, but push gzipped file to S3
+                netcdf_file.publish_type = PipelineFilePublishType.HARVEST_ONLY
+                netcdf_file_gz_collection.set_publish_types(PipelineFilePublishType.UPLOAD_ONLY)  # to set to all files in the collection
+
+            else:
+                netcdf_file.publish_type = PipelineFilePublishType.NO_ACTION
+                netcdf_file_gz_collection.publish_type = PipelineFilePublishType.NO_ACTION
+
+            # deletion of the previous file
+            if remove_previous_version:
+                previous_file_name = os.path.basename(previous_file_object.keys()[0])
+                file_to_delete = PipelineFile(previous_file_name,
+                                              is_deletion=True)
+                self.file_collection.add(file_to_delete)
+                file_to_delete.dest_path = self.dest_path(previous_file_name)
+                file_to_delete.publish_type = PipelineFilePublishType.DELETE_ONLY
+
+    def get_previous_version_object(self, filepath):
+        destination = self.dest_path(filepath)
+        destination_no_creation_date = re.sub('_C-[0-9]{8}T[0-9]{6}Z.*$', '', destination)
+
+        results = self.state_query.query_storage(destination_no_creation_date)
+        if results:
+            return results
+        else:
+            return None
+
+    def get_previous_version_creation_date(self, filepath):
+        """
+        find matching old netcdf_file.gz object already on s3 by stripping the creation date of the new file object path
+        :return: old creation date if exists
+                 returns None if no previous file was found
+        """
+
+        results = self.get_previous_version_object(filepath)
+        if results:
+            return get_creation_date(results.keys()[0])
+        else:
+            return None
+
+    @staticmethod
+    def dest_path(filepath):
+        file_basename = os.path.basename(filepath)
+        gsla_type = get_gsla_type(filepath)
+
+        if GSLA_REGEX_YEARLY.match(file_basename):
```

Comment:
This is true but it means testing against a string (here 'product_type' or os.path.join('DM00', 'yearfiles') ) and repeating that information twice in the code rather than checking for boolean validation against one regexp only

Reply:
Ok.